src/extensions/agent_modules/shared/mosquitto_utils/src/adu_mosquitto_utils.c

```diff
@@ -24,6 +24,11 @@
 // keep this last to avoid interfering with system headers
 #include <aduc/aduc_banned.h>
 
+static bool not_a_dash(char c)
```

Comment:
seems oddly specific (and not really reusable) to create a "not a dash" method.  just add an inline lambda at the source.

Reply:
oh crap, it's C code.

Reply:
yeah, I'd definitely use a lambda if c++